### ISSUE TITLE
update Gateway API docs

### ISSUE THREAD

**Issue by skriss** [projectcontour/contour#4468]
Docs for Gateway API need to be updated to account for changes over the last release, primarily covering the provisioner.

Specifics:
- Dynamic provisioning (using the provisioner) vs. static provisioning ("normal" Contour install)
- Static provisioning: configuring a controller name vs. a gateway ref
- Contour's constraints on listeners (HTTP/HTTPS/TLS only, 2 ports max, but can have multiple listeners differentiated on hostname)
  - [ ] https://github.com/projectcontour/contour/pull/4462#discussion_r850637270
- Dynamic provisioning:
  - GatewayClasses & Params
  - Reconciliation model
  - ...

**Comment by sunjayBhatia** [comment]
I would add the getting started guide as well since the gateway API site refers to this page, maybe add the provisioner above the operator option (or remove the operator option completely): https://projectcontour.io/guides/gateway-api/

**Comment by Rycieos** [comment]
> Contour's constraints on listeners (HTTP/HTTPS/TLS only, 2 ports max, but can have multiple listeners differentiated on hostname) 

This part did not make it into the docs. I only discovered this limitation after trying to instantiate a Gateway, and received an error message saying as such.

Not related to the missing docs, but this limitation is very disappointing, as having HTTPS and TLS endpoints on the same Load Balancer IP was the feature I was most excited by in the Gateway API, and I thought I had found a project that could deliver that. I can't find any issue in this project about supporting both HTTPRoutes and TLSRoutes on the same Gateway; does one exist already?
